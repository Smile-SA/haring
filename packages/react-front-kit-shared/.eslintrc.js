module.exports = {
  extends: ['custom'],
  parserOptions: {
    project: ['./tsconfig.json'],
    tsconfigRootDir: __dirname,
  },
  root: true,
};
