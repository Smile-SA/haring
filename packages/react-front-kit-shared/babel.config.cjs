const { babelConfig } = require('test');

module.exports = babelConfig;
