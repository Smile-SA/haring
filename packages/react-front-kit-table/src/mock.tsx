/* eslint-disable react-refresh/only-export-components */
export * from './Components/Table/Table.mock';
export * from './Components/TableGridView/TableGridView.mock';
