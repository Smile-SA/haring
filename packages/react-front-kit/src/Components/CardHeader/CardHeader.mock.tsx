import { Note } from '@phosphor-icons/react';

export const icon = (
  <Note size={38} style={{ marginTop: '12px' }} weight="thin" />
);

export const title = (
  <>
    <h2 style={{ fontSize: '18px', margin: 0 }}>Élément de texte</h2>
    <p style={{ margin: '0' }}>sous-texte</p>
  </>
);
