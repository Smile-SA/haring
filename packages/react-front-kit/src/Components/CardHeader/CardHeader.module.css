.container {
  display: flex;
  gap: 16px;
}

.root {
  background-color: var(--mantine-primary-color-9);
  box-shadow: 0 0 32px rgb(0, 0, 0, 0.22);
  color: var(--mantine-color-white);
  padding: 24px 32px;
}
