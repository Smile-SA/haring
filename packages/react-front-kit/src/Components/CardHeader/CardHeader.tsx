'use client';

import type { BoxProps } from '@mantine/core';
import type { ReactElement, ReactNode } from 'react';

import { Box } from '@mantine/core';

import classes from './CardHeader.module.css';

export interface ICardHeaderProps extends BoxProps {
  children: ReactNode;
  leftSection?: ReactNode;
  rightSection?: ReactNode;
}

export function CardHeader(props: ICardHeaderProps): ReactElement {
  const { children, leftSection, rightSection, ...rootProps } = props;

  return (
    <Box className={classes.root} {...rootProps}>
      <div className={classes.container}>
        {Boolean(leftSection) && <div>{leftSection}</div>}
        <div>{children}</div>
        {Boolean(rightSection) && <div>{rightSection}</div>}
      </div>
    </Box>
  );
}
