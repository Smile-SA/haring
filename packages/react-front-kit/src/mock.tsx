/* eslint-disable react-refresh/only-export-components */

export * from './Components/ResponsiveTabs/ResponsiveTabs.mock';
export * from './Components/SidebarMenu/SidebarMenu.mock';
export * from './Components/SwitchableView/SwitchableView.mock';
export * from './Components/Thumbnail/Thumbnail.mock';
export * from './Components/ThumbnailGrid/ThumbnailGrid.mock';
export * from './Layouts/FoldableColumnLayout/FoldableColumnLayout.mock';
