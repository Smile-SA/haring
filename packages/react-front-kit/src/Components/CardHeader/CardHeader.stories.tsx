import type { Meta, StoryObj } from '@storybook/react';

import { Badge, Button, Card, Group, Image, Text } from '@mantine/core';

import { CardHeader as Cmp } from './CardHeader';
import { icon, title } from './CardHeader.mock';

const meta = {
  component: Cmp,
  decorators: [
    (Story) => (
      <div>
        <Card padding={0} radius="lg" shadow="sm" withBorder>
          <Story />
          <Card.Section>
            <Image
              alt="Norway"
              height={160}
              src="https://images.unsplash.com/photo-1527004013197-933c4bb611b3?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=720&q=80"
            />
          </Card.Section>
          <Card.Section m="24px 32px">
            <Group justify="apart" mb="xs" mt="md">
              <Text fw={500}>Norway Fjord Adventures</Text>
              <Badge color="pink" variant="light">
                On Sale
              </Badge>
            </Group>

            <Text color="dimmed" size="sm">
              With Fjord Tours you can explore more of the magical fjord
              landscapes with tours and activities on and around the fjords of
              Norway
            </Text>

            <Button
              color="blue"
              fullWidth
              mb="xl"
              mt="md"
              radius="md"
              variant="light"
            >
              Book classic tour now
            </Button>
          </Card.Section>
        </Card>
      </div>
    ),
  ],
  tags: ['autodocs'],
  title: '3-custom/Components/CardHeader',
} satisfies Meta<typeof Cmp>;

export default meta;
type IStory = StoryObj<typeof meta>;

export const CardHeaderWithLeftSection: IStory = {
  args: {
    children: title,
    leftSection: icon,
  },
};

export const CardHeader: IStory = {
  args: {
    children: title,
  },
};
