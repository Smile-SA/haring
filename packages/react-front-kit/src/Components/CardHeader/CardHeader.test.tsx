import { renderWithProviders } from '@smile/react-front-kit-shared/test-utils';

import { CardHeader } from './CardHeader';
import { icon, title } from './CardHeader.mock';

describe('CardHeader', () => {
  it('matches snapshot', () => {
    const { container } = renderWithProviders(
      <CardHeader leftSection={icon}>{title}</CardHeader>,
    );
    expect(container).toMatchSnapshot();
  });
});
