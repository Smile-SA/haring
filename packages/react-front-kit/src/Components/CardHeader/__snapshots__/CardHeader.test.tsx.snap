// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardHeader matches snapshot 1`] = `
<div>
  <div
    class="mantine-ii3g4p"
  >
    <div
      class="mantine-n0q97n"
    >
      <div>
        <svg
          fill="currentColor"
          height="38"
          style="margin-top: 12px;"
          viewBox="0 0 256 256"
          width="38"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M92,96a4,4,0,0,1,4-4h64a4,4,0,0,1,0,8H96A4,4,0,0,1,92,96Zm4,36h64a4,4,0,0,0,0-8H96a4,4,0,0,0,0,8Zm32,24H96a4,4,0,0,0,0,8h32a4,4,0,0,0,0-8ZM220,48V156.69a11.9,11.9,0,0,1-3.52,8.48l-51.31,51.32a11.93,11.93,0,0,1-8.48,3.51H48a12,12,0,0,1-12-12V48A12,12,0,0,1,48,36H208A12,12,0,0,1,220,48ZM48,212H156V160a4,4,0,0,1,4-4h52V48a4,4,0,0,0-4-4H48a4,4,0,0,0-4,4V208A4,4,0,0,0,48,212Zm158.35-48H164v42.35Z"
          />
        </svg>
      </div>
      <div>
        <h2
          style="font-size: 18px; margin: 0px;"
        >
          Text element
        </h2>
        <p
          style="margin: 0px;"
        >
          subtext
        </p>
      </div>
    </div>
  </div>
</div>
`;
